lib/rules/disallow-yoda-conditions.js

```diff
@@ -34,21 +43,22 @@ module.exports = function() {};
 
 module.exports.prototype = {
 
-    configure: function(options) {
+    configure: function(operators) {
+        var isTrue = operators === true;
+
         assert(
-            options === true,
-            this.getOptionName() + ' option requires a true value or should be removed'
+            Array.isArray(operators) || isTrue,
+            this.getOptionName() + ' option requires array or true value'
         );
-        this._operatorIndex = {
-            '==': true,
-            '===': true,
-            '!=': true,
-            '!==': true,
-            '>': true,
-            '<': true,
-            '>=': true,
-            '<=': true
-        };
+
+        if (isTrue) {
+            operators = ['==', '===', '!=', '!==', '>', '<', '>=', '<='];
```

Comment:
I'm not sure about this. As to me they are all Yoda-style. And still I can't see enough reasons to break backwards compatibility.